ui/models.py

```diff
@@ -66,24 +69,85 @@ def s3_key(self):
             user=self.creator.id, uuid=str(self.s3_subkey), base=basename, ext=extension)
         return newkey
 
+    @property
```

Comment:
I don't think this should be a `property` if it's having a side effect, particularly because it is getting invoked in the template. Also, it'd probably be best to separate the data pull from the rest. I'd probably refactor the refresh into an api method and call the refresh in the view code.